packages/cli/generators/controller/index.js

```diff
@@ -181,6 +181,12 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
         when: this.artifactInfo.idType === undefined,
         default: 'number',
       },
+      {
+        type: 'confirm',
+        name: 'idRequired',
+        message: 'Is the id required when creating an instance?',
```

Comment:
I am not sure if this question makes it clear what are the implications of `yes` and `no` answers. Every model instance must eventually have some id value set in the database, thus one can argue that the id property is always required (in a way).

I think it may be better to instead ask the user if the `id` property is generated by the database. When the id is generated, then it must be excluded from `create` requests - I thinks this implication is easier to understand.

```suggestion
        message: 'Is the id (primary key) value generated by the database?',
```

If you decide to accept my suggestion, then please rename the question/answer name from `idRequired` to something more sensible.

Reply:
@bajtos Hmm, I have a different opinion on this, I am thinking of being consistent with the LB 3 options.

See https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html

`required`: Implies "Whether a value for the property is required. If true, then adding or updating a model instance requires a value for the property." and this is my purpose of adding the prompt.

`generated`: "The generated property indicates the ID will be automatically generated by the database. If true, the connector decides what type to use for the auto-generated key. For relational databases, such as Oracle or MySQL, it defaults to number. If your application generates unique IDs, set it to false." It describes a more complicated scenario than whether a property is required or not in the request payload...

Reply:
I see your point.

There is still one missing piece though: the id property is special, there are three possible variants:

1. The id is required when creating an instance. (It's always generated by the client, useful e.g. for offline-first.) TS type to use: `T` (e.g. `Product`)
2. The id is not required but can be provided when creating an instance. (The client can decide whether they want to supply their own id or let the DB create one.) TS type to use: `T & {id?: <number|string>}` (I am not entirely sure about the right type)
3. The id must not be provided and will be always generated by the database. TS type to use: `Omit<T, 'id'>`.

By default, LB apps are configured for the option 3 - the `id` property is not allowed in "create" requests.

That's why I find your prompt confusing, because to me, it selects between the first and second variant, but then emits TypeScript type for the third variant.

Reply:
@bajtos I see, maybe let me change the prompt to be "Is the id omitted when creating an instance?"

- Yes: use `Omit<T, 'id'>`
- No: use `T`

For the 2nd scenario, I think it could be same as 1 if `id` is defined as optional property in the model, like:

```ts
// scenario 1
@model()
class Product{
  @property()
   id: string;
}

// scenario 2
@model()
class Product{
  @property()
   id?: string;
}
```